lib/nerve/service_watcher.rb

```diff
@@ -154,6 +164,21 @@ def check_and_report
 
       report_succeeded = true
       if is_up != @was_up
+        if ! @rate_limiter.consume
+          log.warn "nerve: service #{@name} throttled (rate limiter enabled: #{@rate_limit_enabled})"
+          statsd.increment('nerve.watcher.throttled', tags: ["service_name:#{@name}", "rate_limit_enabled:#{@rate_limit_enabled}"])
+
+          # When the request is throttled, ensure that the status is reported
+          # the next time around.
+          # This returns `nil` (instead of `false`) in order to avoid crashing
+          # the service watcher because of repeated failures. `nil` specifically
+          # reports that the requests were throttled.
```

Comment:
maybe `unless shadow_mode`, I prefer that since rate limit is technically enabled (doing calculation), it just runs in shadow mode.